backend/src/main/java/com/github/boltzmann/biokiste/backend/controller/AppUserController.java

```diff
@@ -0,0 +1,33 @@
+package com.github.boltzmann.biokiste.backend.controller;
+
+import com.github.boltzmann.biokiste.backend.model.AppUserDetails;
+import com.github.boltzmann.biokiste.backend.security.model.AppUser;
+import com.github.boltzmann.biokiste.backend.security.service.AppUserLoginDetailsService;
+import com.github.boltzmann.biokiste.backend.service.AppUserDetailsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.security.Principal;
+
+@RestController
+@RequestMapping("/api/user/")
+public class AppUserController {
+
+    private final AppUserLoginDetailsService appUserLoginDetailsService;
+    private final AppUserDetailsService appUserDetailsService;
+
+    @Autowired
+    public AppUserController(AppUserLoginDetailsService appUserLoginDetailsService, AppUserDetailsService appUserDetailsService) {
+        this.appUserLoginDetailsService = appUserLoginDetailsService;
+        this.appUserDetailsService = appUserDetailsService;
+    }
+
+    @GetMapping("/me")
+    public AppUserDetails getLoggedInUserDetails(Principal principal){
+        AppUser appUser = this.appUserLoginDetailsService
+                .getUserByName(principal.getName());
+        return appUserDetailsService.getUserDetails(appUser);
```

Comment:
Könnte man den Principal-Namen als ID für AppUserDetails nutzen? Dann bräuchte man in der Regel den AppUser (inkl. Passwort) gar nicht von der Datenbank laden, stimmt's?

Reply:
Wenn ich den Principal Namen as ID nehme, dann muss ich zwar nicht mehr getUserByName machen, aber irgendwoher benötige ich die Details ja und irgendwo müssen die persistiert und dann auch geladen werden.